# **Proyecto II**
## Erwin Barriga

In [1]:
import pandas as pd

### CONJUNTO DE DATOS 1

#### **Personality Prediction System**

In [5]:
df1 = pd.concat([pd.read_csv("./Datos 1/train.csv"), pd.read_csv("./Datos 1/test.csv")]) 

# Display all columns
pd.set_option('display.max_column', None)

print(df1.shape)
df1.head()

(1024, 8)


Gender  Age  openness  neuroticism  conscientiousness  agreeableness  \
0    Male   17         7            4                  7              3   
1    Male   19         4            5                  4              6   
2  Female   18         7            6                  4              5   
3  Female   22         5            6                  7              4   
4  Female   19         7            4                  6              5   

   extraversion Personality (Class label)  
0             2               extraverted  
1             6                   serious  
2             5                dependable  
3             3               extraverted  
4             4                    lively

#### 1) Fuente de datos

`https://www.kaggle.com/`

#### 2) Breve descripción de datos

`Dataset basado en el modelo de personalidad de los 5 grandes`

#### 3) ¿Cuál es el objetivo?

`Predecir el tipo de personalidad según la información del puntaje`

#### 4) ¿Este es un problema de clasificación o regresión?
`Clasificación`

#### 5) ¿Cuántas características hay?
`7` 

#### 6) ¿Cuántas filas de datos hay?

`1024`

#### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

`Tengo 5 categorias para el resultado de clasificación`

In [7]:
df1["Personality (Class label)"].value_counts(normalize=True)

serious        0.306641
extraverted    0.221680
responsible    0.162109
dependable     0.155273
lively         0.154297
Name: Personality (Class label), dtype: float64

### CONJUNTO DE DATOS 2

#### **Predicting Pulsar Star**

In [11]:
df2 = pd.concat([pd.read_csv("./Datos 2/pulsar_data_train.csv"), pd.read_csv("./Datos 2/pulsar_data_test.csv")]) 

# Display all columns
pd.set_option('display.max_column', None)

print(df2.shape)
df2.head()

(17898, 9)


Mean of the integrated profile  \
0                       121.156250   
1                        76.968750   
2                       130.585938   
3                       156.398438   
4                        84.804688   

    Standard deviation of the integrated profile  \
0                                      48.372971   
1                                      36.175557   
2                                      53.229534   
3                                      48.865942   
4                                      36.117659   

    Excess kurtosis of the integrated profile  \
0                                    0.375485   
1                                    0.712898   
2                                    0.133408   
3                                   -0.215989   
4                                    0.825013   

    Skewness of the integrated profile   Mean of the DM-SNR curve  \
0                            -0.013165                   3.168896   
1                             3.388719                   2.399666   
2                            -0.297242                   2.743311   
3                            -0.171294                  17.471572   
4                             3.274125                   2.790134   

    Standard deviation of the DM-SNR curve  \
0                                18.399367   
1                                17.570997   
2                                22.362553   
3                                      NaN   
4                                20.618009   

    Excess kurtosis of the DM-SNR curve   Skewness of the DM-SNR curve  \
0                              7.449874                      65.159298   
1                              9.414652                     102.722975   
2                              8.508364                      74.031324   
3                              2.958066                       7.197842   
4                              8.405008                      76.291128   

   target_class  
0           0.0  
1           0.0  
2           0.0  
3           0.0  
4           0.0

#### 1) Fuente de datos

`https://www.kaggle.com/`

#### 2) Breve descripción de datos

`Dataset de observaciones astrómicas de estrellas de neutrones`

#### 3) ¿Cuál es el objetivo?

`Predecir si la estrella en cuestión es o no un Pulsar`

#### 4) ¿Este es un problema de clasificación o regresión?
`Clasificación`

#### 5) ¿Cuántas características hay?
`8`

#### 6) ¿Cuántas filas de datos hay?

`17898`

#### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

`El target está muy desbalanceado y hay algunos datos faltantes`

In [12]:
df2["target_class"].value_counts(normalize=True)

0.0    0.907966
1.0    0.092034
Name: target_class, dtype: float64

In [15]:
df2.isna().sum()

 Mean of the integrated profile                     0
 Standard deviation of the integrated profile       0
 Excess kurtosis of the integrated profile       2502
 Skewness of the integrated profile                 0
 Mean of the DM-SNR curve                           0
 Standard deviation of the DM-SNR curve          1702
 Excess kurtosis of the DM-SNR curve                0
 Skewness of the DM-SNR curve                     869
target_class                                     5370
dtype: int64